ray-curator/ray_curator/stages/deduplication/semantic/kmeans.py

```diff
@@ -0,0 +1,371 @@
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Literal
+
+import cupy as cp
+import numpy as np
+
+from ray_curator.backends.base import WorkerMetadata
+from ray_curator.stages.base import CompositeStage, ProcessingStage
+from ray_curator.stages.deduplication.io_utils import DeduplicationIO
+from ray_curator.stages.file_partitioning import FilePartitioningStage
+from ray_curator.stages.resources import Resources
+from ray_curator.stages.text.embedders.utils import create_list_series_from_1d_or_2d_ar
+from ray_curator.tasks import FileGroupTask, _EmptyTask
+
+from .utils import break_parquet_partition_into_groups, get_array_from_df
+
+if TYPE_CHECKING:
+    import cudf
+
+import time
+
+import torch
+from loguru import logger
+
+# Column names
+L2_DIST_TO_CENT_COL = "l2_dist_to_cent"
+COSINE_DIST_TO_CENT_COL = "cosine_dist_to_cent"
+
+FILETYPE_TO_EXTENSIONS = {
+    "parquet": [".parquet"],
+    "jsonl": [".jsonl", ".json"],
+}
```

Comment:
Could this be useful in a shared `utils.py` or `constants.py` file?

Reply:
Related to https://github.com/NVIDIA-NeMo/Curator/pull/912#discussion_r2285837712 will address it together

Reply:
Somewhat related. I've been bitten by the default extension for `FilePartitioningTask` being `.jsonl`. Would prefer the default behavior there to not filter and each reader (jsonl, parquet etc.) can set whatever extension they want to filter on

Reply:
> prefer the default behavior there to not filter and each reader (jsonl, parquet etc.) can set whatever extension
Agreed in this case I guess in @VibhuJawa 's PR we want to amke sure JsonlReader exposes `file_extensions` attribbutes that is then passed to `FilePartitioningStage`

> This mapping might be useful as a general shared constant in utils outside dedup
I can see if above is what we're doing then any reader can just get the defaults unless specified